wp_api/src/date.rs

```diff
@@ -0,0 +1,72 @@
+use chrono::{DateTime, Utc};
+use serde::{Deserialize, Serialize};
+use std::str::FromStr;
+use wp_serde_helper::wp_utc_date_format;
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Eq)]
+pub struct WpGmtDateTime(#[serde(with = "wp_utc_date_format")] pub DateTime<Utc>);
+
+impl WpGmtDateTime {
+    pub fn from_timestamp(seconds: i64) -> Self {
+        let date_time =
+            DateTime::<Utc>::from_timestamp(seconds, 0).unwrap_or(DateTime::<Utc>::UNIX_EPOCH);
+        Self(date_time)
+    }
```

Comment:
What do you think about adding a few unit tests for this function with different kinds of values (zero, positive, negative)?

Reply:
I don't mind it, but would it be useful? We'd be testing the underlying implementation for `DateTime::<Utc>` and I generally don't add those tests as I feel they don't add any value.

Btw, I am not too thrilled about the `.unwrap_or(DateTime::<Utc>::UNIX_EPOCH)` part, but with `custom_type!` I think we either do this or panic. Out of them, I'd prefer us to panic so that if there are any issues, we'd be aware of it. Otherwise, this may only show up as a support request. I kept your implementation from that temporary PR and I was thinking of bringing it up in the PR description.

Reply:
I thought ensuring we always use "second" as the unit might be useful. Plus, adding test for some distant past and future dates might be helpful to ensure `WpGmtDateTime` continues to support a large range of dates.

The only reason that the parsing can fail is that the date is super distant. I think the 1970 date is an okay fallback for this parsing error because the original super distant is probably a mistake anyway. At bare minimal, we should not panic when parsing them.

Reply:
Addressed in https://github.com/Automattic/wordpress-rs/pull/556.